latex-syntax/src/main/java/org/xwiki/contrib/latex/internal/FigureTool.java

```diff
@@ -42,4 +49,60 @@
      * @return true if the figure caption macro is the last content inside the figure macro content or false otherwise
      */
     boolean isFigureCaptionLast(Block figureCaptionBlock);
+
+    /**
+     * Returns the environment to use for the current caption. By default, "table" or "figure" are returned but
+     * extensions can allow for different environments. Note that different environments mights require different
+     * parameters (see {@link #getFigureParameter(FigureBlock)}), or might not support captions in their content (see
+     * {@link #displayFigureCaption(FigureCaptionBlock)}).
+     *
+     * @param figureBlock the figure block for which to get the environment name
+     * @return the name of the environment (e.g., "figure", "block", or another value in case of customization by an
+     *     extension)
+     * @since 1.21
+     */
+    @Unstable
```

Comment:
Following the recent conversation on the forum, it's probably best to drop the unstable annotation for now.